pkg/sql/set_zone_config.go

```diff
@@ -491,6 +491,15 @@ func (n *setZoneConfigNode) startExec(params runParams) error {
 				"cannot remove default zone")
 		}
 
+		// Secondary tenants are not allowed to set zone configurations on any named
+		// zones other than RANGE DEFAULT.
+		if !params.p.execCfg.Codec.ForSystemTenant() &&
+			zonepb.IsNamedZoneID(targetID) &&
+			targetID != keys.RootNamespaceID {
```

Comment:
nit: Instead of checking against `keys.RootNamespaceID`, check against `zonepb.DefaultZoneName` instead.

```
if !params.p.execCfg.Codec.ForSystemTenant() {
  if name, found := zonepb.NamedZonesByID(targetID); found && name != zonepb.DefaultZoneName {
    // ...
  } 
}
```